### DISCUSSION TITLE
agent限制cpu资源后采集数据有异常

### DISCUSSION THREAD

**Discussion by zcl1115** [harmonycloud/kindling#624] [agent collector]
agent在一台机器上正常需要接近2core，然后显示的采集结果是正常的，没有error请求结果，延迟计算也正常，日志也没有异常打印，但是如果使用cgroup限制最大为1core之后，就出现了一些error请求统计，延迟计算也从原来的几十ms到几分钟，同时日志大量打印Fail to parse dns request这个关键字的信息，想问问有没有遇到类似的情况，需要怎么改配置适配

**Comment by dxsup** [comment]
v0.9.0版本引入了一个变更导致目前CPU使用会出现异常高的情况，将很快在v0.9.1版本修复。

除了上述这个异常外，在请求较多的环境中，agent的CPU消耗也会较高，如果CPU被限制，则数据无法被完整处理，会导致计算结果出现问题。因此要得到准确的计算结果，不能限制CPU。

要想降低CPU使用量，可以通过调整配置文件关闭部分功能。我们之前已经在其他discussion中讨论过了那些能够关闭的功能，如果这些都关闭但CPU使用量仍然较高，那只能通过profile检查代码然后针对性优化性能了。

**Reply by zcl1115** [reply]
好的呢，能问下是什么样的变更吗，因为这边只是将0.9.0的部分代码合并到了这边的工程中

**Reply by dxsup** [reply]
为了定时清理内存引入了一个新的功能，可以看一下main分支上最新的commit，这个commit修复了问题。
